### PR TITLE
Fix #1463 by adding keyspace filter as suggested

### LINKED ISSUE #1463: GET /indexes does not filter by keyspace
In `IndexesResource`:
```java
           BoundQuery query =
                restDB
                    .queryBuilder()
                    .select()
                    .column(columns)
                    .from(SYSTEM_SCHEMA, INDEXES_TABLE)
                    .where("table_name", Predicate.EQ, tableName)
                    .allowFiltering(true)
```
is wrong. If two tables with the same name exist in different keyspaces, all of their indexes will be returned.
It should be instead:
```java
           BoundQuery query =
                restDB
                    .queryBuilder()
                    .select()
                    .column(columns)
                    .from(SYSTEM_SCHEMA, INDEXES_TABLE)
                    .where("keyspace_name", Predicate.EQ, keyspaceName)
                    .where("table_name", Predicate.EQ, tableName)
```

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds missing filter for keyspace in "get all indexes"; extends existing test case to cover it (fails without fix, passes with it)

**Which issue(s) this PR fixes**:
Fixes #1463

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
@olim7t simple fix as per your suggestion.

**Review by mpenick — APPROVED** [review]

File: testing/src/main/java/io/stargate/it/http/RestApiv2SchemaTest.java [hunk 1]
```diff
@@ -1186,6 +1206,17 @@ public void indexListAll() throws IOException {
...
+    // but also that the other table with same name (but in diff keyspace) does not
+    body =
+        RestUtils.get(
+            authToken,
+            String.format(
+                "%s/v2/schemas/keyspaces/%s/tables/%s/indexes",
+                restUrlBase, altKeyspaceName, tableName),
+            HttpStatus.SC_OK);
+    data = objectMapper.readValue(body, new TypeReference<List<Map<String, Object>>>() {});
+    assertThat(data).isEqualTo(Arrays.asList());
```
Comment (mpenick) [inline testing/src/main/java/io/stargate/it/http/RestApiv2SchemaTest.java:1219]: Curious, why `.isEqualTo(Array.asList())` vs `.isEmpty()`? Is there an advantage?

**Comment by tatu-at-datastax** [comment]
Quick note: Stargate V2 listAllFilters works without changes due to the way it is implemented -- was not having this issue.

File: testing/src/main/java/io/stargate/it/http/RestApiv2SchemaTest.java (hunk 1 above)
Comment (mpenick) [reply testing/src/main/java/io/stargate/it/http/RestApiv2SchemaTest.java:1219]: Awesome idea! Thanks for sharing.

Comment (tatu-at-datastax) [reply testing/src/main/java/io/stargate/it/http/RestApiv2SchemaTest.java:1219]: Glad this was useful! I have learned many things via code reviews here, esp. constructs others use that are new to me.

**Pull request merged**